File: src/TinyBlog.DataServices/Entities/Post.cs
using MongoDB.Bson.Serialization.Attributes;
using System;

namespace TinyBlog.DataServices.Entities
{
    [BsonIgnoreExtraElements]
    public class Post : BaseEntity
    {
        [BsonElement("title")]
        public string Title { get; set; }

        [BsonElement("linkText")]
        public string LinkText { get; set; }

        [BsonElement("previewText")]
        public string PreviewText { get; set; }

        [BsonElement("fullText")]
        public string FullText { get; set; }

        [BsonElement("publishedAt")]
        public DateTime PublishedAt { get; set; }

        [BsonElement("tags")]
        public string[] Tags { get; set; }

        [BsonElement("isPublished")]
        public bool IsPublished { get; set; }
    }
}

File: src/TinyBlog.Web/Scripts/testcafe/types/post.ts
import * as mongoose from 'mongoose';

export interface PostDomain extends mongoose.Document {
    title: string;
    linkText: string;
    previewText: string;
    fullText: string;
    publishedAt: Date;
    tags: string[];
    isPublished: boolean;
}

const PostSchema: mongoose.Schema = new mongoose.Schema({
    title: { type: String, required: true },
    linkText: { type: String, required: true, unique: true },
    previewText: { type: String, required: true },
    fullText: { type: String, required: true },
    publishedAt: { type: Date },
    tags: { type: [String] },
    isPublished: { type: Boolean, required: true }
});

export const Post = mongoose.model<PostDomain>('posts', PostSchema);